src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -418,7 +415,7 @@ public ComponentInfo getComponentInfo(String project, String component, String p
 
   private ComponentInfo parseComponentInfo(String json, ComponentKind kind) throws IOException {
     JSonParser parser = new JSonParser(Serialization.json().readTree(json));
-    return parser.parseDescribeComponentInfo(kind, isPodmanPresent);
+    return parser.parseDescribeComponentInfo(kind, isPodmanPresent());
```

Comment:
why modify this variable to a method call each time a compoenent is parsed ? this is time consuming for no reason, has podman is present or not for the whole time the extension is instanciated.

Reply:
the reason is the following: the check was executed in the constructor and it's a heavy operation. All the heavy operations that are in the constructor of OdoCli get executed when the tree structure requests the odo binary. This then lead to the tree displaying "IndicatorCancelled" when the UI was waiting for too long for these operations to be finished.
In this particular case, the instance variable isn't even needed. There's only 1 place where it's value is needed. That's where I replaced the instance var by a direct call.